src/stream.rs

```diff
@@ -160,36 +122,91 @@ where
     }
 }
 
-impl<S1, S2> Stream for WeakSelect<S1, S2>
-where
-    S1: Stream,
-    S2: Stream<Item = S1::Item, Error = S1::Error>,
-{
-    type Item = S1::Item;
-    type Error = S1::Error;
+pub use new_futures::{SelectWeak, VecStreamExt};
 
-    fn poll(&mut self) -> Poll<Option<S1::Item>, S1::Error> {
-        let (a, b) = if self.flag {
-            (
-                &mut self.stream2 as &mut dyn Stream<Item = _, Error = _>,
-                &mut self.stream1 as &mut dyn Stream<Item = _, Error = _>,
-            )
-        } else {
-            (
-                &mut self.stream1 as &mut dyn Stream<Item = _, Error = _>,
-                &mut self.stream2 as &mut dyn Stream<Item = _, Error = _>,
-            )
-        };
-        self.flag = !self.flag;
+mod new_futures {
```

Comment:
Why is this in an internal module? It's not wrong, just wondering why the additional cognitive overhead.

Reply:
Wanted to leave global import from `futures01` and use `futures03` imports, will be removed in future on removing `take_until`.